### PR TITLE
Ui components

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by benma** [comment]
This should be easy to review - only the header sections changed.

File: src/ui/components/scroll_through_all_variants.c [hunk 1]
```diff
@@ -12,16 +12,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <string.h>
+#include "scroll_through_all_variants.h"
+#include "../event.h"
+#include "button.h"
```
Comment (conte91) [inline src/ui/components/scroll_through_all_variants.c]: `#include` order - `button.h` < `event.h`.

File: src/ui/components/confirm_button.c [hunk 2]
```diff
@@ -15,7 +15,8 @@
 #include "confirm_button.h"
 #include "../event.h"
 #include "../event_handler.h"
-#include "ui_components.h"
+#include "confirm_gesture.h"
+#include "icon_button.h"
```
[the remark this answers is not in the document]
Comment (benma) [reply src/ui/components/confirm_button.c]: We enforce clang-format, which sorts the groups alphabetically.

File: src/ui/components/confirm_transaction.c [hunk 3]
```diff
@@ -13,7 +13,11 @@
 // limitations under the License.
 
 #include "confirm_transaction.h"
-#include "ui_components.h"
+#include "confirm_button.h"
```
[the remark this answers is not in the document]
Comment (benma) [reply src/ui/components/confirm_transaction.c]: The header belonging to the current file should always be on top. If part of a group, clang-format even considers this when formatting the file.

File: src/ui/components/scroll_through_all_variants.c (hunk 1 above)
Comment (benma) [reply src/ui/components/scroll_through_all_variants.c]: But `.` < `b` :P

File: src/ui/components/confirm_transaction.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (conte91) [reply src/ui/components/confirm_transaction.c]: ja - see my edit :)

**Pull request merged**